pkg/api/v1/resource/helpers.go

```diff
@@ -64,6 +65,46 @@ func GetResourceRequestQuantity(pod *v1.Pod, resourceName v1.ResourceName) resou
 	return requestQuantity
 }
 
+// GetResourceAllocatedQuantity finds and returns the allocated quantity for a specific resource.
+func GetResourceAllocatedQuantity(pod *v1.Pod, resourceName v1.ResourceName) resource.Quantity {
+	allocatedQuantity := resource.Quantity{}
+
+	switch resourceName {
+	case v1.ResourceCPU:
+		allocatedQuantity = resource.Quantity{Format: resource.DecimalSI}
+	case v1.ResourceMemory, v1.ResourceStorage, v1.ResourceEphemeralStorage:
+		allocatedQuantity = resource.Quantity{Format: resource.BinarySI}
+	default:
+		allocatedQuantity = resource.Quantity{Format: resource.DecimalSI}
+	}
+
+	for _, container := range pod.Spec.Containers {
+		if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
+			rQuantity := cs.AllocatedResources[resourceName]
```

Comment:
Is there a reason to read the allocated resources from the status? Would it be preferable to directly check the allocation manager since that is the source of truth for pod allocations? My concern is that I'm not sure if there are cases where the status could be lagging behind?